src/main/java/com/minecrafttas/mctcommon/Configuration.java

```diff
@@ -5,40 +5,75 @@
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.InvalidPropertiesFormatException;
-import java.util.Map;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Properties;
 
+import com.minecrafttas.mctcommon.registry.AbstractRegistry;
+import com.minecrafttas.mctcommon.registry.Registerable;
 
 /**
  * A <i>very</i> simple configuration class
+ * 
  * @author Scribble
- *
  */
 
-public class Configuration {
-	
+public class Configuration extends AbstractRegistry<com.minecrafttas.mctcommon.Configuration.ConfigOptions> {
+
 	private File file;
-	
+
 	private Properties properties;
 
 	private String comment;
-	
+
 	public Configuration(String comment, File configFile) {
+		super("Configuration", new LinkedHashMap<>());
+
 		file = configFile;
 		this.comment = comment;
+	}
+
+	protected final List<ConfigOptions> configRegistry = new ArrayList<>();
+	
+	@Override
+	public void register(ConfigOptions registryObject) {
+		if(registryObject == null) {
+			return;
+		}
+		
+		if(configRegistry.contains(registryObject)) {
+			return;
+		}
+		
+		configRegistry.add(registryObject);
+	}
+	
+	@Override
+	public void unregister(ConfigOptions registryObject) {
+		if (registryObject == null) {
+			return;
+		}
+
+		if (!configRegistry.contains(registryObject)) {
+			return;
+		}
 		
-		if(file.exists()) {
-			properties = load();
+		configRegistry.remove(registryObject);
+	}
+
+	public void load() {
+		if (file.exists()) {
+			properties = loadInner();
 		}
-		if(properties == null || !file.exists()) {
+		if (properties == null || !file.exists()) {
 			properties = generateDefault();
 			save();
-		}
+		}		
```

Comment:
can you please turn on your save actions to fix these whitespaces automatically, haha